src/v/redpanda/admin_server.cc

```diff
@@ -1636,18 +1636,18 @@ admin_server::kafka_transfer_leadership_handler(
 
     co_return co_await _partition_manager.invoke_on(
       *shard,
-      ss::coroutine::lambda(
         [ntp = std::move(ntp), target, this, req = std::move(req)](
-          cluster::partition_manager& pm) mutable
-        -> ss::future<ss::json::json_return_type> {
+          cluster::partition_manager& pm) mutable {
             auto partition = pm.get(ntp);
             if (!partition) {
                 throw ss::httpd::not_found_exception();
```

Comment:
Should this use `make_exception_future`?

Reply:
i think that @graphcareful added a futurize_invoke wrapper around all of this so that we could do this safely since it's a pretty common pattern. in any case, i agree it's not the most idiomatic. should probably be changed (or even better, more of these converted into factored out coroutines).

Reply:
Oh, it's wrapped in an `invoke_on`, which eventually catches exceptions and converts them, so this is completely fine regardless. I was thinking it throws past the fiber, so any intermediate handling would be ignored.